src/odemis/driver/zeiss.py

```diff
@@ -483,20 +518,37 @@ def terminate(self):
             self._pos_poll.cancel()
             self._pos_poll = None
 
-    def _updatePosition(self, raw_pos=None):
+    def _getStagePosition(self) -> Tuple[Dict[str, float], bool]:
         """
-        update the position VA
-        raw_pos (dict str -> float): the position in mm (as received from the SEM)
+        Read the stage position, using either GetStagePosition of GetStagePosition6 where the first supports 3
+        linear axes and the latter 6 axes including rx and rz rotational axes.
+        :return position dict str -> float: (in m/rad) with the axes supported in .axes 
+                is_moving (bool)
         """
-        if raw_pos is None:
-            x, y, z, _ = self.parent.GetStagePosition()
+        pos = {}
+
+        if len(self.axes) > 3:
+            x, y, z, t, r, m, is_moving = self.parent.GetStagePosition6()
+            pos["rx"] = math.radians(t)
+            pos["rz"] = math.radians(r)
+            pos["m"] = m * 1e-3
         else:
-            x, y, z = raw_pos["x"], raw_pos["y"], raw_pos["z"]
+            x, y, z, is_moving = self.parent.GetStagePosition()
+
+        pos["x"] = x * 1e-3
+        pos["y"] = y * 1e-3
+        pos["z"] = z * 1e-3
```

Comment:
You should only copy the axes to pos **if** they are in self.axes!
For instance, if "z" is not in self.axes, pos shouldn't contain it.

Note that one option would be to do the code as you have it here, and just after add a line to remove the axes which are not present.